runtime/src/iree/builtins/ukernel/arch/x86_64/mmt4d_x86_64_avx512_bf16.c

```diff
@@ -43,18 +41,28 @@ static inline __m512 iree_mm512_dpbf16_ps_broadcast_rhs(
 #endif
 
 static inline void
-iree_uk_mmt4d_tile_bf16bf16f32_1x16x2_to_16x16x2_x86_64_avx512_bf16(
+iree_uk_mmt4d_tile_bf16bf16fXX_1x16x2_to_16x16x2_x86_64_avx512_bf16(
     void* IREE_UK_RESTRICT out_tile, const void* IREE_UK_RESTRICT lhs_panel,
     const void* IREE_UK_RESTRICT rhs_panel,
-    const iree_uk_mmt4d_params_t* params, int M0) {
+    const iree_uk_mmt4d_params_t* params, iree_uk_type_t acc_type, int M0) {
+  IREE_UK_ASSERT(acc_type == IREE_UK_TYPE_FLOAT_32 ||
+                 acc_type == IREE_UK_TYPE_BFLOAT_16);
   IREE_UK_ASSERT(M0 >= 1 && M0 <= 16 && iree_uk_is_po2_u32(M0));
-  float* IREE_UK_RESTRICT out_ptr = out_tile;
   const iree_uk_uint16_t* IREE_UK_RESTRICT lhs_ptr = lhs_panel;
   const iree_uk_uint16_t* IREE_UK_RESTRICT rhs_ptr = rhs_panel;
   __m512 acc[16];
   if (params->flags & IREE_UK_FLAG_MMT4D_ACCUMULATE) {
-    for (int i = 0; i < M0; ++i) {
-      acc[i] = _mm512_loadu_ps(out_ptr + i * 16);
+    if (acc_type == IREE_UK_TYPE_FLOAT_32) {
+      float* IREE_UK_RESTRICT out_ptr = out_tile;
+      for (int i = 0; i < M0; ++i) {
+        acc[i] = _mm512_loadu_ps(out_ptr + i * 16);
+      }
+    } else {
+      iree_uk_uint16_t* IREE_UK_RESTRICT out_ptr = out_tile;
+      for (int i = 0; i < M0; ++i) {
+        __m256i loaded = _mm256_loadu_si256((const __m256i*)(out_ptr + i * 16));
+        acc[i] = _mm512_cvtpbh_ps(*(const __m256bh*)&loaded);
```

Comment:
I see errors here building on my Windows machine:

```
[build] [739/7322   9% :: 11.296] Building C object runtime\src\iree\builtins\ukernel\arch\x86_64\CMakeFiles\iree_builtins_ukernel_arch_x86_64_x86_64_avx512_bf16.objects.dir\mmt4d_x86_64_avx512_bf16.c.obj
[build] FAILED: runtime/src/iree/builtins/ukernel/arch/x86_64/CMakeFiles/iree_builtins_ukernel_arch_x86_64_x86_64_avx512_bf16.objects.dir/mmt4d_x86_64_avx512_bf16.c.obj 
[build] ccache C:\PROGRA~2\MICROS~1\2022\Preview\VC\Tools\MSVC\1431~1.311\bin\Hostx64\x64\cl.exe  /nologo  -external:ID:\dev\projects\iree -external:ID:\dev\projects\iree-build -external:ID:\dev\projects\iree\runtime\src\iree\.. -external:ID:\dev\projects\iree-build\runtime\src\iree\.. -external:W0 /DWIN32 /D_WINDOWS /Z7 /O2 /Ob1  -MD /DWIN32_LEAN_AND_MEAN /DNOMINMAX /D_USE_MATH_DEFINES /D_CRT_SECURE_NO_WARNINGS /D_CRT_SECURE_CPP_OVERLOAD_STANDARD_NAMES /GR- /bigobj /Zc:preprocessor /std:c11 /W3 /wd4200 /wd4018 /wd4146 /wd4244 /wd4267 /wd4005 /wd4065 /wd4141 /wd4624 /wd5105 /arch:AVX512 -std:c11 /showIncludes /Foruntime\src\iree\builtins\ukernel\arch\x86_64\CMakeFiles\iree_builtins_ukernel_arch_x86_64_x86_64_avx512_bf16.objects.dir\mmt4d_x86_64_avx512_bf16.c.obj /Fdruntime\src\iree\builtins\ukernel\arch\x86_64\CMakeFiles\iree_builtins_ukernel_arch_x86_64_x86_64_avx512_bf16.objects.dir\ /FS -c D:\dev\projects\iree\runtime\src\iree\builtins\ukernel\arch\x86_64\mmt4d_x86_64_avx512_bf16.c
[build] D:\dev\projects\iree\runtime\src\iree\builtins\ukernel\arch\x86_64\mmt4d_x86_64_avx512_bf16.c(64): warning C4013: '_mm512_cvtpbh_ps' undefined; assuming extern returning int
[build] D:\dev\projects\iree\runtime\src\iree\builtins\ukernel\arch\x86_64\mmt4d_x86_64_avx512_bf16.c(64): error C2440: '=': cannot convert from 'int' to '__m512'
```

CI looks green though. Guess my machine has avx512 and the CI doesn't, or something like that?